### ISSUE TITLE
Memory leaking from webapp; cannot shutdown normally

### ISSUE THREAD

**Issue by GoogleCodeExporter** [sagarswathi/h2database#603]
```
Please find the report of this bug on 
https://groups.google.com/forum/#!topic/h2-database/aMenH0lCCf0

This bug means my application is leaking memory on each restart of the 
application, which cannot be tolerated in production. Therefor this matter is 
urgent to me.

As this is impeding my ability to stop the application normally (I must kill 
the server for all threads to stop), it is limiting my workflow considerably.

I do not have a workaround (other than switching to another in memory 
database). 

```

Original issue reported on code.google.com by `friso.vr...@gmail.com` on 2 Mar 2015 at 11:05

**Comment by GoogleCodeExporter** [comment]
```
Hi,

Do you know, how much memory does it leak each time?
```

Original comment by `thomas.t...@gmail.com` on 5 Mar 2015 at 7:55

**Comment by GoogleCodeExporter** [comment]
```
Hi, 

No, I don't know and it doesn't matter all that much. The main problem I'm 
facing now is that the redeploy of the application doesn't go well. basically, 
the old version doesn't stop. This forces me to shut down the server upon 
redeploy, impacting all other applications on the server. This is unacceptable 
for our operations and this forces me to use another database if there is no 
fix/workaround.

Groeten,

Friso
```

Original comment by `friso.vr...@gmail.com` on 17 Mar 2015 at 6:34

**Comment by GoogleCodeExporter** [comment]
```
> basically, the old version doesn't stop

OK, that sounds bad. This needs to be fixed.
```

Original comment by `thomas.t...@gmail.com` on 18 Mar 2015 at 6:52
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
Original comment by `thomas.t...@gmail.com` on 18 Mar 2015 at 7:30
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
This should be fixed in the trunk now.
```

Original comment by `thomas.t...@gmail.com` on 23 Mar 2015 at 8:53

**Comment by GoogleCodeExporter** [comment]
```
Great! Thanks
```

Original comment by `friso.vr...@gmail.com` on 23 Mar 2015 at 10:17

**Comment by GoogleCodeExporter** [comment]
```
Should be fixed with version 1.4.187
```

Original comment by `thomas.t...@gmail.com` on 10 Apr 2015 at 6:54
- Changed state: **Fixed**
